src/main/java/org/owasp/benchmark/score/parsers/TestResults.java

```diff
@@ -0,0 +1,267 @@
+		return this.benchmarkVersion;
+	}
+	
+	public ToolType getToolType() {
+	    return toolType;
+	}
+	
+	public boolean isCommercial() {
+	    return isCommercial;
+	}
+	
+	public void put( TestCaseResult tcr ) {
+		List<TestCaseResult> results = map.get( tcr.getNumber() );
+		if ( results == null ) {
+			results = new ArrayList<TestCaseResult>();
+			map.put( tcr.getNumber(),  results );
+		}
+		results.add( tcr );
+	}
+	
+	public List<TestCaseResult> get( int tn ) {
+		return map.get( tn );
+	}
+
+	public Set<Integer> keySet() {
+		return map.keySet();
+	}
+    
+	/**
+	 * Get the name of the tool. e.g., "IBM AppScan"
+	 * @return Name of the tool.
+	 */
+    public String getTool() {
+        return this.toolName;
+    }
+
+	/**
+	 * Get the name of the tool and its version together. e.g., "IBM AppScan v4.2". But if the tool is commercial,
+	 * and its in anonymous mode then don't include the version number as that could give away the tool.
+	 * @return Name of the tool.
+	 */
+    public String getToolNameAndVersion() {
+    	if (!anonymous && this.toolVersion != null && !"".equals(this.toolVersion) && 
+    		!(BenchmarkScore.anonymousMode && this.isCommercial)) {
+            return this.toolName + " v" + this.toolVersion;
+    	}
+        return this.toolName;
+    }
+    
+    public String getToolVersion() {
+        return toolVersion;
+    }
+	
+    /**
+     * Sets the name of the tool. e.g., "HP Fortify"
+     * @param tool - Name of the tool.
+     */
+	public void setTool( String toolName ) {
+	    this.toolName = toolName;
+	}
+
+	/**
+	 * This method anonymizes the tool name based on the type of tool it is and the count of previous 
+	 * tools of the same type that also have been anonymized.
+	 */
+	public void setAnonymous() {
+		//System.out.println("Anonymizing tool: " + this.getTool() + " which is of type: " + getToolType());
+		this.anonymous = true;
+
+		switch (getToolType()) {
+			case SAST : {
+				if (nextCommercialSAST_ToolNumber < 10) {
+					this.setTool("SAST-0" + nextCommercialSAST_ToolNumber++);
+				} else this.setTool("SAST-" + nextCommercialSAST_ToolNumber++);
+				break;
+			}
+			case DAST : {
+				if (nextCommercialDAST_ToolNumber < 10) {
+					this.setTool("DAST-0" + nextCommercialDAST_ToolNumber++);
+				} else this.setTool("DAST-" + nextCommercialDAST_ToolNumber++);				
+				break;
+			}
+			case IAST : {
+				if (nextCommercialIAST_ToolNumber < 10) {
+					this.setTool("IAST-0" + nextCommercialIAST_ToolNumber++);
+				} else this.setTool("IAST-" + nextCommercialIAST_ToolNumber++);				
+			}
+		}
+	}
+	
+	public void setToolVersion( String version ) {
+	    this.toolVersion = version;
+	}
+
+    public String getTime() {
+        return time;
+    }
+    
+    public void setTime( String elapsed ) {
+        this.time = elapsed;
+    }
+    
+    public void setTime ( File f ) {
+		String filename = f.getName();
+		String time = filename.substring(filename.lastIndexOf('-')+1, filename.lastIndexOf('.'));
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

If the lastIndexOf() search on line 179 does not find the substring and returns -1, the call to substring() on line 179 will throw IndexOutOfBoundsException.